src/extrapolation/filled.jl

```diff
@@ -54,6 +54,17 @@ end
     return ret
 end
 
+@inline function Interpolations.gradient(etp::FilledExtrapolation{T}, x::Vararg{Number,N}) where {T,N}
+    itp = parent(etp)
+    if checkbounds(Bool, itp, x...)
+        gradient(itp, x...)
+    else
+        e = eltype(itp)
+        ptype = promote_type(e, typeof(etp.fillvalue))
+        return SVector{ndims(etp), ptype}(zeros(ptype, ndims(etp)))
+    end
```

Comment:
The type instability (`@inferred` failures) comes from the fact that this if-else block might return different types, hence @SomTambe you might need to double-check if `ptype` is appropriate. See https://docs.julialang.org/en/v1/manual/performance-tips/#Write-%22type-stable%22-functions for more explanation.

cc: @mkitti